test-data/unit/check-classes.test

```diff
@@ -4223,3 +4223,12 @@ class C(Any):
         reveal_type(self.bar().__name__) # E: Revealed type is 'builtins.str'
 [builtins fixtures/type.pyi]
 [out]
+
+[case testClassDecoratorIsTypeChecked]
+from typing import Callable
+def decorate(x: int) -> Callable[[type], type]:  # N: "decorate" defined here
+    ...
+
+@decorate(y=17)  # E: Unexpected keyword argument "y" for "decorate"
+class A:
+    pass
```

Comment:
The PR is trivial, but I would add few more test cases for various kinds of errors, like non-existing name/attribute, calling non-callable, too many args, forward reference in a decorator type, etc.